pkg/network/setup/network.go

```diff
@@ -46,11 +46,18 @@ func NewVMNetworkConfigurator(vmi *v1.VirtualMachineInstance, cacheCreator cache
 	return newVMNetworkConfiguratorWithHandlerAndCache(vmi, &netdriver.NetworkUtilsHandler{}, cacheCreator)
 }
 
-func (v VMNetworkConfigurator) getPhase1NICs(launcherPID *int) ([]podNIC, error) {
+func (v VMNetworkConfigurator) getPhase1NICs(launcherPID *int, networksToPlug ...v1.Network) ([]podNIC, error) {
 	var nics []podNIC
 
-	for i := range v.vmi.Spec.Networks {
-		nic, err := newPhase1PodNIC(v.vmi, &v.vmi.Spec.Networks[i], v.handler, v.cacheCreator, launcherPID)
+	if len(v.vmi.Spec.Domain.Devices.Interfaces) == 0 {
+		return nics, nil
+	}
+
+	if len(networksToPlug) == 0 {
+		networksToPlug = v.vmi.Spec.Networks
```

Comment:
I prefer using defaults since we're already feeding the VMI to this function - it is confusing imo to give it a VMI (that has networks) and a networks list. 

If you have really strong feelings about this, I would appreciate if this could happen in a follow-up.

Reply:
I have a problem with this variable being named as is, because it gives a false interpretation as if these interfaces are to be hotplug. But they are surely not when this input is empty.

Beyond that, I am willing to follow up on the function itself, which is now odd in my view. Functions are suppose to do only one thing, but here it does two different things, depending on the input. If the hotplug network list is empty it process only it, if it is not empty it processes the VMI networks. This is forcing one function to do two things and I think it is not a pattern we want to see.

Reply:
I don't think this does two things, it does one: plug networks. 

It plugs either the networks you give it, or, by default, the networks defined within the VMI spec. 

Do you mind sorting this out as a follow up ?

Reply:
Thank you; I appreciate it.